### ISSUE TITLE
Silence logging

### ISSUE THREAD

**Issue by TemplarVolk** [mongo-dart/mongo_dart#141]
It seems there's no way to stop mongo_dart from printing to the console. Or at least none that i can find.

I'm printing other things to the console but are buried with query logs.

Not sure if I'm missing something or not.

**Comment by TemplarVolk** [comment]
I should say i didnt have this problem back when i was using 0.3.2

**Comment by thosakwe** [comment]
Import `package:logging`, and set `hierarchicalLoggingEnabled` to `true` as early as possible in your application. This will solve your problem.

I wish this were `true` by default, but it's not, which is often quite annoying.

**Comment by TemplarVolk** [comment]
Sorry but this made no difference

**Comment by thosakwe** [comment]
Can you paste your output? Also, see if you can include the `hierarchicalLoggingEnabled` line as the very first line in your `main()`.

**Comment by TemplarVolk** [comment]
For example everytime it makes queries
```bash
[FINE] {MongoMessageTransformer} - 2019-01-07 02:08:48.778124: MongoReplyMessage(ResponseTo:117, cursorId: 0, numberReturned:1, responseFlags:8, {...}})
[FINE] {Connection} - 2019-01-07 02:08:48.778190: MongoReplyMessage(ResponseTo:117, cursorId: 0, numberReturned:1, responseFlags:8, {...}})
[FINE] {Connection} - 2019-01-07 02:08:48.778231: Completing MongoReplyMessage(ResponseTo:117, cursorId: 0, numberReturned:1, responseFlags:8, {...}})
[FINE] {Connection} - 2019-01-07 02:08:48.778942: Execute MongoUpdateMessage(118, data.users, {...}})
```
(i replaced the contents of the queries with `...` cause i cant share them)

**Comment by thosakwe** [comment]
I don't think this is a problem with mongo_dart, actually. This package doesn't add any additional logic; it just creates instances of `Logger`.

Do you have an `onRecord.listen(...)` call anywhere? You don't want to call `Logger.root.onRecord(...)` if `hierarchicalLoggingEnabled` is `true`.

Instead, call it directly on your logger, i.e.:

```dart
hierarchicalLoggingEnabled = true;
Logger('my_logger').onRecord.listen(...);
```

**Comment by TemplarVolk** [comment]
I ended up not setting `hierarchicalLoggingEnabled` to `true` since it doesn't have to be for the root level.

> I don't think this is a problem with mongo_dart, actually. This package doesn't add any additional logic; it just creates instances of Logger.

Usually in other packages there is a way to change the logging level of these `Logger` instances.

**Comment by thosakwe** [comment]
You can send a PR to make the loggers on MongoMessageHandler, etc. public.
